app/code/Magento/Checkout/Model/ShippingInformationManagement.php

```diff
@@ -185,7 +185,7 @@ public function saveAddressInformation(
         } catch (\Exception $e) {
             $this->logger->critical($e);
             throw new InputException(
-                __('The shipping information was unable to be saved. Verify the input data and try again.')
+                __('The shipping information was unable to be saved. Error: "%1"', $e->getMessage())
```

Comment:
Its bad idea to show any type of exception to user. 
For example, it can say `The shipping information was unable to be saved. Error: Cannot connect to Mysql Server with user: user1 and password: pass`

Reply:
I'm not sure MySQL could throw that kind of exception. I got some other improvements to exception messages approved and today I faced that one..so I decided to improve it too. I believe what I've done is safe.

Reply:
That would be good improvement if it will not show message for all types of exceptions, but only from some specific, for instance only for LocalizedException . What do you think about it?

Reply:
↑ LocalizedException is something that can be actually shown to the user.
↑ `\Exception` should be logged and generic message provided to the user.